--- ./sources/nft_rental.move
+++ ./sources/nft_rental.move
@@ -46 +46 @@
-    struct Rentables has drop {}
+    public struct Rentables has drop {}
@@ -50 +50 @@
-    struct Rented has store, copy, drop { id: ID }
+    public struct Rented has store, copy, drop { id: ID }
@@ -54 +54 @@
-    struct Listed has store, copy, drop { id: ID }
+    public struct Listed has store, copy, drop { id: ID }
@@ -57 +57 @@
-    struct Promise {
+    public struct Promise {
@@ -68 +68 @@
-    struct Rentable<T: key + store> has store {
+    public struct Rentable<T: key + store> has store {
@@ -78 +78 @@
-    struct RentalPolicy<phantom T> has key, store {
+    public struct RentalPolicy<phantom T> has key, store {
@@ -91 +91 @@
-    struct ProtectedTP<phantom T> has key, store {
+    public struct ProtectedTP<phantom T> has key, store {
@@ -207 +207 @@
-        coin: Coin<SUI>,
+        mut coin: Coin<SUI>,
@@ -213 +213 @@
-        let rentable = take_from_bag<T, Listed>(renter_kiosk, Listed { id: item_id });
+        let mut rentable = take_from_bag<T, Listed>(renter_kiosk, Listed { id: item_id });
--- ./tests/tests.move
+++ ./tests/tests.move
@@ -25,2 +25,2 @@
-    struct T has key, store {id: UID}
-    struct WITNESS has drop {}
+    public struct T has key, store {id: UID}
+    public struct WITNESS has drop {}
@@ -31 +31 @@
-        let scenario= test_scenario::begin(RENTER);
+        let mut scenario= test_scenario::begin(RENTER);
@@ -43 +43 @@
-        let scenario= test_scenario::begin(RENTER);
+        let mut scenario= test_scenario::begin(RENTER);
@@ -57 +57 @@
-        let scenario= test_scenario::begin(RENTER);
+        let mut scenario= test_scenario::begin(RENTER);
@@ -76 +76 @@
-        let scenario= test_scenario::begin(RENTER);
+        let mut scenario= test_scenario::begin(RENTER);
@@ -101 +101 @@
-        let scenario= test_scenario::begin(RENTER);
+        let mut scenario= test_scenario::begin(RENTER);
@@ -124 +124 @@
-        let scenario= test_scenario::begin(RENTER);
+        let mut scenario= test_scenario::begin(RENTER);
@@ -143 +143 @@
-            let kiosk = test_scenario::take_shared_by_id<Kiosk>(test, renter_kiosk_id);
+            let mut kiosk = test_scenario::take_shared_by_id<Kiosk>(test, renter_kiosk_id);
@@ -159 +159 @@
-        let scenario= test_scenario::begin(RENTER);
+        let mut scenario= test_scenario::begin(RENTER);
@@ -187 +187 @@
-        let scenario= test_scenario::begin(RENTER);
+        let mut scenario= test_scenario::begin(RENTER);
@@ -215 +215 @@
-        let scenario= test_scenario::begin(RENTER);
+        let mut scenario= test_scenario::begin(RENTER);
@@ -252 +252 @@
-        let scenario= test_scenario::begin(RENTER);
+        let mut scenario= test_scenario::begin(RENTER);
@@ -274 +274 @@
-            let kiosk = test_scenario::take_shared_by_id<Kiosk>(test, renter_kiosk_id);
+            let mut kiosk = test_scenario::take_shared_by_id<Kiosk>(test, renter_kiosk_id);
@@ -291 +291 @@
-        let scenario= test_scenario::begin(BORROWER);
+        let mut scenario= test_scenario::begin(BORROWER);
@@ -324 +324 @@
-        let scenario= test_scenario::begin(BORROWER);
+        let mut scenario= test_scenario::begin(BORROWER);
@@ -355 +355 @@
-        let scenario= test_scenario::begin(BORROWER);
+        let mut scenario= test_scenario::begin(BORROWER);
@@ -388 +388 @@
-        let scenario= test_scenario::begin(BORROWER);
+        let mut scenario= test_scenario::begin(BORROWER);
@@ -420 +420 @@
-        let scenario= test_scenario::begin(BORROWER);
+        let mut scenario= test_scenario::begin(BORROWER);
@@ -455 +455 @@
-        let scenario= test_scenario::begin(BORROWER);
+        let mut scenario= test_scenario::begin(BORROWER);
@@ -482 +482 @@
-            let kiosk = test_scenario::take_shared_by_id<Kiosk>(test, borrower_kiosk_id);
+            let mut kiosk = test_scenario::take_shared_by_id<Kiosk>(test, borrower_kiosk_id);
@@ -498 +498 @@
-        let scenario= test_scenario::begin(BORROWER);
+        let mut scenario= test_scenario::begin(BORROWER);
@@ -534 +534 @@
-        let scenario= test_scenario::begin(BORROWER);
+        let mut scenario= test_scenario::begin(BORROWER);
@@ -561 +561 @@
-            let kiosk = test_scenario::take_shared_by_id<Kiosk>(test, borrower_kiosk_id);
+            let mut kiosk = test_scenario::take_shared_by_id<Kiosk>(test, borrower_kiosk_id);
@@ -581 +581 @@
-        let scenario= test_scenario::begin(BORROWER);
+        let mut scenario= test_scenario::begin(BORROWER);
@@ -618 +618 @@
-        let scenario= test_scenario::begin(BORROWER);
+        let mut scenario= test_scenario::begin(BORROWER);
@@ -657 +657 @@
-        let scenario= test_scenario::begin(BORROWER);
+        let mut scenario= test_scenario::begin(BORROWER);
@@ -693 +693 @@
-        let scenario= test_scenario::begin(BORROWER);
+        let mut scenario= test_scenario::begin(BORROWER);
@@ -698 +698 @@
-        let clock = clock::create_for_testing(test_scenario::ctx(test));
+        let mut clock = clock::create_for_testing(test_scenario::ctx(test));
@@ -728 +728 @@
-        let scenario= test_scenario::begin(RENTER);
+        let mut scenario= test_scenario::begin(RENTER);
@@ -734 +734 @@
-        let clock = clock::create_for_testing(test_scenario::ctx(test));
+        let mut clock = clock::create_for_testing(test_scenario::ctx(test));
@@ -765 +765 @@
-        let scenario= test_scenario::begin(BORROWER);
+        let mut scenario= test_scenario::begin(BORROWER);
@@ -770 +770 @@
-        let clock = clock::create_for_testing(test_scenario::ctx(test));
+        let mut clock = clock::create_for_testing(test_scenario::ctx(test));
@@ -804 +804 @@
-        let scenario= test_scenario::begin(BORROWER);
+        let mut scenario= test_scenario::begin(BORROWER);
@@ -809 +809 @@
-        let clock = clock::create_for_testing(test_scenario::ctx(test));
+        let mut clock = clock::create_for_testing(test_scenario::ctx(test));
@@ -840 +840 @@
-        let scenario= test_scenario::begin(BORROWER);
+        let mut scenario= test_scenario::begin(BORROWER);
@@ -845 +845 @@
-        let clock = clock::create_for_testing(test_scenario::ctx(test));
+        let mut clock = clock::create_for_testing(test_scenario::ctx(test));
@@ -878 +878 @@
-        let scenario= test_scenario::begin(RENTER);
+        let mut scenario= test_scenario::begin(RENTER);
@@ -890 +890 @@
-            let kiosk = test_scenario::take_shared_by_id<Kiosk>(test, kiosk_id);
+            let mut kiosk = test_scenario::take_shared_by_id<Kiosk>(test, kiosk_id);
@@ -920 +920 @@
-            let kiosk = test_scenario::take_shared_by_id<Kiosk>(scenario, kiosk_id);
+            let mut kiosk = test_scenario::take_shared_by_id<Kiosk>(scenario, kiosk_id);
@@ -933 +933 @@
-            let kiosk = test_scenario::take_shared_by_id<Kiosk>(scenario, kiosk_id);
+            let mut kiosk = test_scenario::take_shared_by_id<Kiosk>(scenario, kiosk_id);
@@ -954 +954 @@
-            let kiosk = test_scenario::take_shared_by_id<Kiosk>(scenario, kiosk_id);
+            let mut kiosk = test_scenario::take_shared_by_id<Kiosk>(scenario, kiosk_id);
@@ -969 +969 @@
-            let kiosk = test_scenario::take_shared_by_id<Kiosk>(scenario, kiosk_id);
+            let mut kiosk = test_scenario::take_shared_by_id<Kiosk>(scenario, kiosk_id);
@@ -982 +982 @@
-            let kiosk = test_scenario::take_shared_by_id<Kiosk>(scenario, kiosk_id);
+            let mut kiosk = test_scenario::take_shared_by_id<Kiosk>(scenario, kiosk_id);
@@ -997 +997 @@
-            let kiosk = test_scenario::take_shared_by_id<Kiosk>(scenario, kiosk_id);
+            let mut kiosk = test_scenario::take_shared_by_id<Kiosk>(scenario, kiosk_id);
@@ -1012,3 +1012,3 @@
-            let borrower_kiosk = test_scenario::take_shared_by_id<Kiosk>(scenario, borrower_kiosk_id);
-            let renter_kiosk = test_scenario::take_shared_by_id<Kiosk>(scenario, renter_kiosk_id);
-            let rental_policy = test_scenario::take_shared<RentalPolicy<T>>(scenario);
+            let mut borrower_kiosk = test_scenario::take_shared_by_id<Kiosk>(scenario, borrower_kiosk_id);
+            let mut renter_kiosk = test_scenario::take_shared_by_id<Kiosk>(scenario, renter_kiosk_id);
+            let mut rental_policy = test_scenario::take_shared<RentalPolicy<T>>(scenario);
@@ -1028 +1028 @@
-            let kiosk = test_scenario::take_shared_by_id<Kiosk>(scenario, kiosk_id);
+            let mut kiosk = test_scenario::take_shared_by_id<Kiosk>(scenario, kiosk_id);
@@ -1040 +1040 @@
-        let kiosk = test_scenario::take_shared_by_id<Kiosk>(scenario, kiosk_id);
+        let mut kiosk = test_scenario::take_shared_by_id<Kiosk>(scenario, kiosk_id);
@@ -1054 +1054 @@
-            let kiosk = test_scenario::take_shared_by_id<Kiosk>(scenario, kiosk_id);
+            let mut kiosk = test_scenario::take_shared_by_id<Kiosk>(scenario, kiosk_id);
@@ -1065,2 +1065,2 @@
-            let borrower_kiosk = test_scenario::take_shared_by_id<Kiosk>(scenario, borrower_kiosk_id);
-            let renter_kiosk = test_scenario::take_shared_by_id<Kiosk>(scenario, renter_kiosk_id);
+            let mut borrower_kiosk = test_scenario::take_shared_by_id<Kiosk>(scenario, borrower_kiosk_id);
+            let mut renter_kiosk = test_scenario::take_shared_by_id<Kiosk>(scenario, renter_kiosk_id);
@@ -1082 +1082 @@
-            let transfer_policy = test_scenario::take_shared<TransferPolicy<T>>(scenario);
+            let mut transfer_policy = test_scenario::take_shared<TransferPolicy<T>>(scenario);
